web/api/v1/api.go

```diff
@@ -574,6 +585,130 @@ func (api *API) alertmanagers(r *http.Request) (interface{}, *apiError, func())
 	return ams, nil, nil
 }
 
+// AlertDiscovery has info for all alerts
+type AlertDiscovery struct {
+	AlertGroups []*AlertGroup `json:"alertGroups"`
+}
+
+// Alert has info for a alert
+type Alert struct {
+	Labels      labels.Labels `json:"labels"`
+	Annotations labels.Labels `json:"annotations"`
+	State       string        `json:"state"`
+	ActiveAt    *time.Time    `json:"activeAt,omitempty"`
```

Comment:
We might want to be a little careful with this one, its semantics may change over time.

Reply:
Yes. Users also depend on these values in ways that aren't correct, these are implementation details more than anything.